starport/cmd/network_chain_init.go

```diff
@@ -0,0 +1,141 @@
+package starportcmd
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/starport/starport/pkg/cliquiz"
+	"github.com/tendermint/starport/starport/pkg/clispinner"
+	"github.com/tendermint/starport/starport/pkg/cosmosaccount"
+	"github.com/tendermint/starport/starport/services/chain"
+	"github.com/tendermint/starport/starport/services/network"
+)
+
+const (
+	flagValidatorAccount = "validator-account"
+)
+
+// NewNetworkChainInit returns a new command to initialize a chain from a published chain ID
+func NewNetworkChainInit() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "init [launch-id]",
+		Short: "Initialize a chain from a published chain ID",
+		Args:  cobra.ExactArgs(1),
+		RunE:  networkChainInitHandler,
+	}
+
+	c.Flags().String(flagValidatorAccount, cosmosaccount.DefaultAccount, "Account for the chain validator")
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagSetYes())
+
+	return c
+}
+
+func networkChainInitHandler(cmd *cobra.Command, args []string) error {
+	nb, s, shutdown, err := initializeNetwork(cmd)
+	if err != nil {
+		return err
+	}
+	defer shutdown()
+
+	// parse launch ID
+	launchID, err := strconv.ParseUint(args[0], 10, 64)
+	if err != nil {
+		return errors.Wrap(err, "error parsing launchID")
+	}
+	if launchID == 0 {
+		return errors.New("launch ID must be greater than 0")
+	}
+
+	// check if the provided account for the validator exists
+	validatorAccount, _ := cmd.Flags().GetString(flagValidatorAccount)
+	if err := checkAccountExist(cmd, validatorAccount); err != nil {
+		return err
+	}
+
+	// if a chain has already been initialized with this launch ID, we ask for confirmation before erasing the directory
+	chainHome, exist, err := network.IsChainHomeExist(launchID)
+	if err != nil {
+		return err
+	}
+	if !getYes(cmd) && exist {
+		prompt := promptui.Prompt{
+			Label: fmt.Sprintf("The chain has already been initialized under: %s. Would you like to overwrite the home directory",
+				chainHome,
+			),
+			IsConfirm: true,
+		}
+		s.Stop()
+		if _, err := prompt.Run(); err != nil {
+			fmt.Println("said no")
+			return nil
+		}
+		s.Start()
```

Comment:
sometimes it spawns two or infinite loop of prompt questions

Reply:
It may be a bug with `promptui`. For me it happens sometimes when the terminal window is too small

Reply:
someone already reported the issue into the repo: https://github.com/manifoldco/promptui/issues/186